### PR TITLE
Steam Authentication System Reworked

### PR DESCRIPTION
**Pull request by zubEpty**
## Summary by CodeRabbit

- **New Features**
  - Introduced a GitHub Actions workflow for automatic updates and publishing to the UPM branch.
  - Added a GitHub Actions workflow for documentation generation and deployment.
  - Implemented a new authentication system extension for Steam in Unity.
  - Provided a Steam authentication editor window for Unity developers.
  - Included a new `SteamProvider` class for Steam authentication logic.

- **Documentation**
  - Added README.md with installation and usage instructions for the Steam authentication extension.
  - Created a CHANGELOG.md for the base authentication package.
  - Included LICENSE.md with the MIT License for the project.

- **Bug Fixes**
  - Adjusted texture import settings for better performance and compatibility.

- **Refactor**
  - Updated Unity package manifest files with new references and namespaces.

- **Style**
  - Enhanced scene settings for improved visual fidelity in Unity's SampleScene.

- **Chores**
  - Added and updated various meta files for Unity asset management.

### CODE REVIEW

File: Assets/Packages/com.studio23.ss2.authsystem.steam/Editor/SteamAuthGeneratorEditorWindow.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            // Create a new instance of your SteamProvider scriptable object
+            SteamProvider steamProvider = ScriptableObject.CreateInstance<SteamProvider>();
+            steamProvider.AppId = _steamAppId;
+
+            // Save the scriptable object to the Resources folder
+            AssetDatabase.CreateAsset(steamProvider, "Assets/Resources/SteamAuthData.asset");
+            AssetDatabase.SaveAssets();
+            AssetDatabase.Refresh();
+            EditorUtility.SetDirty(steamProvider);
+        }
```
[the remark this answers is not in the document]
Comment (Warhammer4000) [reply Assets/Packages/com.studio23.ss2.authsystem.steam/Editor/SteamAuthGeneratorEditorWindow.cs:52]: It should be under AuthSystem Folder 
Assets/Resources/AuthSystem/{name}.asset

File: Assets/Packages/com.studio23.ss2.authsystem/package.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "com.studio23.ss2.authsystem",
+  "version": "1.0",
```
Comment (Warhammer4000) [inline Assets/Packages/com.studio23.ss2.authsystem/package.json:3]: version should be semantic. 1.0.0 For this case lets do it less than 1
0.9.0

File: Assets/Packages/com.studio23.ss2.authsystem/package.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    {
+      "name": "Studio 23",
+      "url": "https://package.openupm.com",
+      "scopes": [
+        "com.studio23.ss2.authsystem"
+      ]
+    }
+  ],
+  "dependencies": {
+    "com.unity.nuget.newtonsoft-json": "3.2.1"
```
Comment (Warhammer4000) [inline Assets/Packages/com.studio23.ss2.authsystem/package.json:21]: why do we need this?

File: Assets/Packages/com.studio23.ss2.authsystem/package.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "com.studio23.ss2.authsystem",
```
Comment (Warhammer4000) [inline Assets/Packages/com.studio23.ss2.authsystem/package.json:2]: wrong

File: Assets/Packages/com.studio23.ss2.authsystem/README.md [hunk 5]
```diff
@@ -0,0 +1 @@
+This system will be used to check user authentication for validating proper digital rights management (DRM) of the game. This is the base package for the system.
```
Comment (N-Evan) [inline Assets/Packages/com.studio23.ss2.authsystem/README.md:1]: While the base auth package is just a placeholder to act as a factory for the other auth modules, a slightly more proper and easily understandable `README.md` is expected.

File: Assets/Packages/com.studio23.ss2.authsystem/Runtime/Data/ProviderBase.cs
Comment (N-Evan) [inline Assets/Packages/com.studio23.ss2.authsystem/Runtime/Data/ProviderBase.cs:1]: Maybe add option functions such as `Logout`?

File: Assets/Packages/com.studio23.ss2.authsystem/package.json (hunk 4 above)
Comment (N-Evan) [reply Assets/Packages/com.studio23.ss2.authsystem/package.json:2]: This is correct in this case as this package.json file belongs to the core package and not the module.

File: Assets/Packages/com.studio23.ss2.authsystem.steam/Runtime/Core/SteamProvider.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        /// </summary>
+        public override void Authenticate()
+        {
+            if (SteamClient.RestartAppIfNecessary(AppId))
+            {
+                SteamClient.Shutdown();
+                Application.Quit();
+            }
+
+            SteamClient.Init(AppId);
```
Comment (N-Evan) [inline Assets/Packages/com.studio23.ss2.authsystem.steam/Runtime/Core/SteamProvider.cs:27]: Do we get any sort of call back from the `Init` function? If so, there should be an exception/error handling when the authentication fails for some reason. (ie. network issues.)

**Review by N-Evan — CHANGES_REQUESTED** [review]
Please remove `AuthSystem` core from the assets folder and instead import it locally as a package while in development. The benefit of doing this locally ensures that any changes made to a locally imported package will automatically be updated on projects that imported it locally. 

The Steam module does not have a `package.json` file. Please ensure that the package.json exists as well as a `CHANGELOG.md` and properly documented `README.md`.

File: Assets/Packages/com.studio23.ss2.authsystem.steam/Runtime/Core/SteamProvider.cs (hunk 6 above)
Comment (zubEpty) [reply Assets/Packages/com.studio23.ss2.authsystem.steam/Runtime/Core/SteamProvider.cs:27]: There's no need to enable callback for exception handling. `Init()` method as it already handles exception on itself whenever it tries to initialize using appId.

**Pull request closed without merging**